packages/template-retail-react-app/scripts/extract-default-messages.js

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env node
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/* eslint @typescript-eslint/no-var-requires: "off" */
+
+/**
+ * This script will extract messages from base template and extended app and output all translations in the extended app
+ * If a file is overridden, it won't extract messages from that file in the base template
+ */
+const {exec} = require('child_process')
+const fs = require('fs')
+const path = require('path')
+const packagePath = path.join(process.cwd(), 'package.json')
+const pkgJSON = JSON.parse(fs.readFileSync(packagePath))
+
+const getAllFilesByExtensions = (dirPath, arrayOfFiles = [], extensions = []) => {
+    const files = fs.readdirSync(dirPath)
+
+    files.forEach(function (file) {
+        if (fs.statSync(path.join(dirPath, file)).isDirectory()) {
+            arrayOfFiles = getAllFilesByExtensions(
+                path.join(dirPath, file),
+                arrayOfFiles,
+                extensions
+            )
+        } else {
+            arrayOfFiles.push(path.join(dirPath, file))
+        }
+    })
+    if (extensions.length) {
+        return arrayOfFiles.filter((filePath) => {
+            const getExtension = path.extname(filePath).replace('.', '')
+            return extensions.includes(getExtension)
+        })
+    }
+
+    return arrayOfFiles
+}
+
+try {
+    const overridesDir = pkgJSON.ccExtensibility?.overridesDir
+    if (!overridesDir) {
+        const command =
+            "formatjs extract 'app/**/*.{js,jsx}' --out-file translations/en-US.json --id-interpolation-pattern [sha512:contenthash:base64:6]"
+        exec(command, (err) => {
+            if (err) {
+                console.error(err)
+            }
+        })
+    } else {
+        const overridesPath = path.join(process.cwd(), pkgJSON.ccExtensibility?.overridesDir)
+        // get all the files in extended app
+        const files = getAllFilesByExtensions(
+            path.join(overridesPath, 'app'),
+            [],
+            ['js', 'jsx', 'ts', 'tsx']
```

Comment:
If we are looking for Typescript files here, then we'll need to update all of the `formatjs` commands to search for those files too.

Reply:
Done.